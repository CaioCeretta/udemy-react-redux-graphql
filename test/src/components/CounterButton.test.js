import { shallow } from "enzyme"
import CounterButton from './CounterButton'


it('expects to render the button component', () => {
  const mockColor = 'blue'

  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot()
})

it('Correctly increments the counter', () => {
  const mockColor = 'blue'

  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate('click')
  wrapper.find('[id="counter"]').simulate('click')
  expect(wrapper.state()).toEqual({ count: 2 })

  wrapper.find('[id="counter"]').simulate('click')

  expect(wrapper.state()).toEqual({ count: 3 })
  
  wrapper.find('[id="counter"]').simulate('keypress')

  expect(wrapper.state()).toEqual({ count: 3 })

  expect(wrapper.props().color).toEqual('blue')
 
   


})