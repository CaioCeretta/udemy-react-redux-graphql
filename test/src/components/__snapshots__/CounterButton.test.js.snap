// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the button component 1`] = `
<button
  color="blue"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
